src/ntpd.c

```diff
@@ -936,6 +942,11 @@ static int ntpd_read (void)
 		return (-1);
 	}
 
+	if (ntohs(ik->status) & STA_NANO) {
```

Comment:
I did some digging, and ntpdc itself always assumes STA_NANO sets resolution (https://github.com/ntp-project/ntp/blob/stable/ntpdc/ntpdc_ops.c#L2946). It's been in ntp since 1999. It's probably safe to drop the check for timex.h and always define STA_NANO as 0x2000.